jujugui/static/gui/src/app/components/deployment-flow/sshkey/sshkey.js

```diff
@@ -30,22 +15,298 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 const AZURE_CLOUD_TYPE = 'azure';
 
 class DeploymentSSHKey extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      addSource: 'github',
+      SSHkeys: [],
+      error: null,
+      buttonDisabled: true
+    };
+  }
+
   /**
-    Handle SSH key content changes.
+    Handle keyboard input to listen for return key.
 
-    @method _onSSHKeyInputBlur
-    @param {Object} evt The blur event.
+    @param {Object} evt The keyboard event.
   */
-  _onSSHKeyInputBlur(evt) {
-    const key = this.refs.sshKey.getValue();
-    this.props.setSSHKey(key);
+  _onKeyUp(evt) {
+    if (evt.which === 13) {
+      this._handleAddMoreKeys(null);
+    }
+    const hasValue = (this.refs.sshKey && this.refs.sshKey.getValue()) ||
+      (this.refs.githubUsername && this.refs.githubUsername.getValue());
+    this.setState({
+      buttonDisabled: hasValue ? false : true
+    });
   }
 
   /**
-    Render the component.
+    Split a Manual Key into its parts.
 
-    @method render
+    @param {String} sshkey
```

Comment:
What format does this key need to be in? The code seems to indicate that it requires a name?